components/locid_transform/src/expander.rs

```diff
@@ -621,26 +663,16 @@ mod tests {
                 LikelySubtagsForLanguageV1Marker::KEY,
                 LikelySubtagsForScriptRegionV1Marker::KEY,
             ],
-            inner: icu_testdata::buffer(),
         };
-        let lc = LocaleExpander::try_new_with_buffer_provider(&provider)
-            .expect("should create with old keys");
-        let mut locale = locale!("zh-CN");
-        assert_eq!(lc.maximize(&mut locale), TransformResult::Modified);
-        assert_eq!(locale, locale!("zh-Hans-CN"));
+        LocaleExpander::try_new_with_any_provider(&provider).expect("should create with old keys");
```

Comment:
Nit (optional): I would prefer if you used provider::BakedProvider to get real data so that you can still verify that the LocaleExpander "actually works" with the old data keys, rather than just asserting that it doesn't fail to build.